### PR TITLE
Set up CI with Azure Pipelines

### PR DESCRIPTION
**Pull request by jbrockmendel**

### CODE REVIEW

**Comment by jbrockmendel** [comment]
@chrisrpatterson you mentioned over in pandas that you may be able to help with azure-pipelines setup.  Can I ping you for help on a much less-popular project?

I think I've gotten most of the config put together.  What I'm still stuck on is the fact that the pipeline is showing Green here on the PR page while there is still a failing test in the most recent log.  Have I missed a config option somewhere?

**Comment by chrisrpatterson** [comment]
The publish test results task will not fail the build based on failed test results alone.  You will need to have the test script return a non-zero exit code to fail the build.

**Comment by jbrockmendel** [comment]
> You will need to have the test script return a non-zero exit code to fail the build.

Thanks for the quick feedback.  If you're not the right person to ask about this, let me know and I'll stop bugging you.

I copied the pandas version as closely as possible, cannot tell where the relevant difference is.  AFACT the pytest call [here](https://github.com/jbrockmendel/sm2/pull/96/files#diff-88d72fad5b93c62b37ac15db1937fb0cR36) should return a non-zero exit code, which should in turn mean a non-zero return code [here](https://github.com/jbrockmendel/sm2/pull/96/files#diff-4a6370d2b0eafd40c563b682e6a7af95R36).  Is it obvious why this works for pandas but not here?

**Comment by jbrockmendel** [comment]
@bashtage any interest in helping troubleshoot this?  I'm kind of hoping that once the config is figured out once it can be shared here/arch/linearmodels/statsmodels

**Comment by bashtage** [comment]
I don't feel a lot of pressure to have higher performance CI, and setup seems to be challenging.

File: ci/script_single.sh [hunk 1]
```diff
@@ -33,6 +33,6 @@ else
 
 fi
 
-RET="$?"
+RET=$?
 
-exit "$RET"
+exit $RET
```
Comment (bashtage) [inline ci/script_single.sh]: WHy not just exit 1 to see if this is the issue?

File: ci/azure/macos.yml [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    - script: |
+        export PATH=$HOME/miniconda3/bin:$PATH
+        ci/incremental/build.sh
+      displayName: 'Build'
+    - script: |
+        export PATH=$HOME/miniconda3/bin:$PATH
+        ci/script_single.sh
+        echo $?
+        ci/script_multi.sh
+        echo $?
```
Comment (jbrockmendel) [inline ci/azure/macos.yml]: > The publish test results task will not fail the build based on failed test results alone. You will need to have the test script return a non-zero exit code to fail the build.

@chrisrpatterson both here and two lines above I'm seeing a `1`, as expected.  Is there somewhere else where a non-zero exit code needs to be returned?

Comment (chrispat) [reply ci/azure/macos.yml]: Can you try adding set -o pipefail as the first line of your script?  I suspect echo returns non-zero and it is the last line.

Comment (jbrockmendel) [reply ci/azure/macos.yml]: @chrisrpatterson thanks for the suggestion.  Added that on line 34 above, no clear difference.
